python/paddle/fluid/incubate/fleet/collective/__init__.py

```diff
@@ -69,6 +125,85 @@ def save_persistables(self, executor, dirname, main_program=None):
 fleet = Collective()
 
 
+class CollectiveOpBasedOptimizer(DistributedOptimizer):
+    """
+    TBA
+    """
+
+    def __init__(self, optimizer, strategy=None):
+        super(CollectiveOpBasedOptimizer, self).__init__(optimizer, strategy)
+
+    def _transpile_program(self, startup_program=None):
+        startup_program = startup_program if startup_program else \
+                          fluid.framework.default_startup_program()
+        worker_endpoints = fleet.worker_endpoints()
+        trainer_id = fleet.worker_index()
+        current_endpoint = fleet.worker_endpoints()[trainer_id]
+        # call transpiler
+        config = dist_transpiler.DistributeTranspilerConfig()
+        config.mode = "collective"
```

Comment:
should be config.mode='nccl2'?

Reply:
collective config is not in develop branch

Reply:
yes, I should wait the collective mode transpiler to merge